### PR TITLE
test(amplify-util-mock): add e2e tests for new transformers

### PR DESCRIPTION
**Pull request by danielleadams**
#### Description of changes
Port non-auth mock util tests over for v2 transformers. (Auth tests to follow in a separate PR.)

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by cjihrig** [comment]
A small generic comment: can we put the v2 tests in their own directory. It will make future maintenance a lot easier, especially for people not working on the initial v2 implementation.

**Comment by danielleadams** [comment]
Ready for merge once it's approved.

**Comment by cjihrig** [comment]
It looks like there are some lgtm bot issues to address.

**Comment by danielleadams** [comment]
@cjihrig fixed the bot alerts. ready for rereview!

File: packages/amplify-util-mock/src/__e2e_v2__/index-transformer.e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,754 @@
...
+test('Test next token with key', async () => {
+  const status = 'PENDING';
+  const createdAt = '2019-06-06T00:01:01.000Z';
+  // createItems
+  await createItem('order1', status, 'item1', '2019-01-06T00:01:01.000Z');
+  await createItem('order2', status, 'item2', '2019-02-06T00:01:01.000Z');
+  await createItem('order3', status, 'item3', '2019-03-06T00:01:01.000Z');
+  await createItem('order4', status, 'item4', '2019-06-06T00:01:01.000Z');
+  // query itemsByCreatedAt with limit of 2
+  // const items = await itemsByCreatedAt(createdAt, { beginsWith: status }, 2);
```
Comment (cjihrig) [inline packages/amplify-util-mock/src/__e2e_v2__/index-transformer.e2e.test.ts:102]: Please remove commented code.

File: packages/amplify-util-mock/src/__e2e_v2__/index-transformer.e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+  expect(createResponseMissingLastSortKey.data.createShippingUpdate).toBeNull();
+  expect(createResponseMissingLastSortKey.errors).toHaveLength(1);
+
+  const createResponseMissingFirstSortKey = await createShippingUpdate({ orderId: '2ndtry', status: 'PENDING', name: '43?' });
+  expect(createResponseMissingFirstSortKey.data.createShippingUpdate).toBeNull();
+  expect(createResponseMissingFirstSortKey.errors).toHaveLength(1);
+
+  await createShippingUpdate({ orderId: 'order1', itemId: 'item1', status: 'PENDING', name: 'name1' });
+  const items = await getShippingUpdates('order1');
+  console.log('items data', items.data);
```
Comment (cjihrig) [inline packages/amplify-util-mock/src/__e2e_v2__/index-transformer.e2e.test.ts:314]: There are a few `console.log()`s in the PR. If we don't need them, can you please remove them.

**Review by cjihrig — APPROVED** [review]
LGTM. Left a few minor comments.

**Pull request merged**